### PR TITLE
AB#15444 Due Date in the past should not be saved

### PR DESCRIPTION
**Pull request by aurelio-aot**
Implemented the following logic:
- Due Date and Decision Date should only be validated when changed
- Form should not be allowed to be saved when Due Date or Decision Date is changed and has validation errors

### CODE REVIEW

File: applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js [hunk 1]
```diff
@@ -85,7 +85,30 @@
     initDatePicker();
 });
 
-
+let dueDateHasChanged = false;
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js:88]: @aurelio-aot - will this not work?

The save becomes this:
function enableResultSaveBtn(element) {    
    if ($("#assessmentResultForm").valid()) {
        $('#saveAssessmentResultBtn').prop('disabled', false);
    }
    else {
        $('#saveAssessmentResultBtn').prop('disabled', true);
    }
}

Not sure i see the need for the individual validation and variables?

Comment (aurelio-aot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js:88]: hi @AndreGAot -- if the Due Date was saved previously and the Due Date is past date now, the code above will prevent the form from getting saved.  In my example below,  I only changed the Likelihood of Funding and not the Due Date but the Due Date is getting validated and preventing the form to be submitted.

image

Comment (AndreGAot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js:88]: okay, i see - catering for legacy data

Comment (AndreGAot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js:88]: Let try refactor this and highlight that it is for legacy data captured, and also do validations in the API

**Review by AndreGAot — APPROVED** [review]
Ready for Dev

**Pull request merged**